pubsub.go

```diff
@@ -284,29 +351,118 @@ func (p *PubsubValueStore) Cancel(name string) (bool, error) {
 		delete(p.subs, name)
 	}
 
+	if cancel, ok := p.cancels[name]; ok {
+		cancel()
+		delete(p.cancels, name)
+	}
+
 	return ok, nil
 }
 
-func (p *PubsubValueStore) handleSubscription(sub *pubsub.Subscription, key string, cancel func()) {
+func (p *PubsubValueStore) handleSubscription(ctx context.Context, sub *pubsub.Subscription, key string) {
 	defer sub.Cancel()
```

Comment:
As mentioned in the above thread https://github.com/libp2p/go-libp2p-pubsub-router/pull/33#discussion_r307090646 this is pretty weird. If for some reason the subscription dies or gets cancelled without being managed by `PubsubValueStore.Cancel(key)` then we can't resubscribe until we call `Cancel(key)`.

It's pretty weird and a bit of an edge case so I'm ok not bothering with this in the current PR if we don't want to, but wanted to make sure this was flagged.

@Stebalien @vyzo thoughts?

Reply:
Well, `handleSubscription` should probably unregister itself on close. But we can fix that in a subsequent patch (could you make an issue?).

Reply:
Filed: libp2p/go-libp2p-pubsub-router#35